test/Confluent.SchemaRegistry.Serdes.IntegrationTests/Tests_Avro/AutoRegisterSchemaDisabled.cs

```diff
@@ -135,6 +135,17 @@ public static void AutoRegisterSchemaDisabled(string bootstrapServers, string sc
                 {
                     producer.ProduceAsync(topic.Name, new Message<string, int> { Key = "test", Value = 112 }).Wait();
                 }
+
+                // config with avro.serializer.use.latest.version == true should also work now.
+                using (var schemaRegistry = new CachedSchemaRegistryClient(new SchemaRegistryConfig { Url = schemaRegistryServers }))
```

Comment:
an important test i think (which you don't have) would be if a newer schema exists in SR, and the client is specifying an older schema, right? ideally, we'd have this test for avro, json and protobuf... i think implementing it for at least one of the cases has value (i recognize it's a bit of a pain).

finally, we have a protobuf problem. the protobuf tests have actually been broken on newer SR versions for some time, and i haven't prioritized looking into it. Since I think your changes actually overlap (the tests may need updating due to your changes), i'd love if you took a look ... (if not, i will get to it).

to run the tests you need .net core 2.1 installed. then `dotnet test -f netcoreapp2.1`. here are the test failures (as occur on your branch, but as i said, some of this is pre-existing).

```
[xUnit.net 00:02:58.3318311]     Confluent.SchemaRegistry.Serdes.IntegrationTests.Tests.ProduceConsumeNestedProtobuf(bootstrapServers: "localhost:9092", schemaRegistryServers: "localhost:8081") [FAIL]
  Failed Confluent.SchemaRegistry.Serdes.IntegrationTests.Tests.ProduceConsumeNestedProtobuf(bootstrapServers: "localhost:9092", schemaRegistryServers: "localhost:8081") [46 s]
  Error Message:
   Assert.Equal() Failure
Expected: 3
Actual:   6
  Stack Trace:
     at Confluent.SchemaRegistry.Serdes.IntegrationTests.Tests.ProduceConsumeNestedProtobuf(String bootstrapServers, String schemaRegistryServers) in /Users/matt/git/confluent-kafka-dotnet/test/Confluent.SchemaRegistry.Serdes.IntegrationTests/Tests_Protobuf/ProduceConsumeNested.cs:line 78
[xUnit.net 00:03:52.0111808]     Confluent.SchemaRegistry.Serdes.IntegrationTests.Tests.ProduceConsumeSchemaManyMessagesProtobuf(bootstrapServers: "localhost:9092", schemaRegistryServers: "localhost:8081") [FAIL]
  Failed Confluent.SchemaRegistry.Serdes.IntegrationTests.Tests.ProduceConsumeSchemaManyMessagesProtobuf(bootstrapServers: "localhost:9092", schemaRegistryServers: "localhost:8081") [46 s]
  Error Message:
   Assert.Equal() Failure
Expected: 1
Actual:   2
  Stack Trace:
     at Confluent.SchemaRegistry.Serdes.IntegrationTests.Tests.ProduceConsumeSchemaManyMessagesProtobuf(String bootstrapServers, String schemaRegistryServers) in /Users/matt/git/confluent-kafka-dotnet/test/Confluent.SchemaRegistry.Serdes.IntegrationTests/Tests_Protobuf/ProduceConsumeSchemaManyMessages.cs:line 77
[xUnit.net 00:05:00.0445875]     Confluent.SchemaRegistry.Serdes.IntegrationTests.Tests.ConsumePartitionEOF(bootstrapServers: "localhost:9092", schemaRegistryServers: "localhost:8081") [FAIL]
  Failed Confluent.SchemaRegistry.Serdes.IntegrationTests.Tests.ConsumePartitionEOF(bootstrapServers: "localhost:9092", schemaRegistryServers: "localhost:8081") [1 s]
  Error Message:
   Assert.NotNull() Failure
  Stack Trace:
     at Confluent.SchemaRegistry.Serdes.IntegrationTests.Tests.ConsumePartitionEOF(String bootstrapServers, String schemaRegistryServers) in /Users/matt/git/confluent-kafka-dotnet/test/Confluent.SchemaRegistry.Serdes.IntegrationTests/Tests_Avro/ConsumePartitionEOF.cs:line 77

Failed!  - Failed:     3, Passed:    23, Skipped:     0, Total:    26, Duration: 4 m 59 s - /Users/matt/git/confluent-kafka-dotnet/test/Confluent.SchemaRegistry.Serdes.IntegrationTests/bin/Debug/netcoreapp2.1/Confluent.SchemaRegistry.Serdes.IntegrationTests.dll (netcoreapp2.1)
```

Reply:
I wasn't able to get my environment set up successfully to run these tests, so it would be great if you could take a look sometime.  In any case, since these tests were already failing, and so as to not overload this PR, I'll create a separate issue for this so we can have a separate PR to fix this issue later.

Reply:
Created https://github.com/confluentinc/confluent-kafka-dotnet/issues/1630

Reply:
sounds good - thanks for looking. i only brought it up on the off chance that as the domain expert, you might see what was wrong with very little effort. i'll take a look myself.